### PR TITLE
Upgrade demo-project to Kedro 0.18

### PR DESCRIPTION
**Pull request by studioswong**
## Description

resolves [this issue.](https://github.com/kedro-org/kedro-viz/issues/806)

This is an initial attempt to upgrade the kedro-viz demo-project (aka modular-spaceflights) to 0.18. 

## Development notes

PART 1: 
I had paired with @MerelTheisenQB on this work, where we upgraded any references of Kedro 0.17 to Kedro 0.18, along as other steps following [this migration guide](https://github.com/kedro-org/kedro/blob/main/RELEASE.md#migration-guide-from-kedro-017-to-018). 

As a summary, we: 

- removed the current `ProjectHooks` class that contains the `register_config_loader` and `register_catalog` hooks definition
- moved references of `CONFIG_LOADER_CLASS` and `CONFIG_LOADER_ARGS` to `settings.pyl` ( due to errors we encountered during our Kedro run, we changed the declaration to a work-around solution as specified by @AntonyMilneQB  [here](https://github.com/kedro-org/kedro/issues/1402#issuecomment-1085771758))
- removed all instances of `Journal` 

The current setup is a WIP where we encountered the following errors, where we deduced that it could be due to [this issue](https://github.com/kedro-org/kedro/issues/1402) with `TemplatedConfigLoader` ( or could possibly be related with loading Jinja template within this project, we are not sure) : 

image

~~@AntonyMilneQB and @datajoely , would love your input into solving the above error given your knowledge of this project and with the known issue with `TemplatedConfigLoader`.~~

On another note, massive thanks to @MerelTheisenQB for pairing with me on this 🤗 

PART 2:
Antony and I got a chance to look into the issue with the config loader and further paired on this, and we made the following changes:

- replaced the reference to `templatedConfigLoader` to a custom one to solve issues with Jinga templating ( to be reverted once a fix for this is out on the latest kedro release)
- updated namespacing for all `params` of modular pipelines within the project
- further improvements, such as the arrangement of the `create_pipeline` functions within the pipeline declarations ( thanks to @AntonyMilneQB 's advice on some Kedro best practices)
-  improvement: bumping `scikit-learn` to a higher version

Massive thanks to @AntonyMilneQB for pairing with me on this 🤗 ( and as always, your detailed explanation into things) 

## QA notes

N/A 

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/827"></a>

### CODE REVIEW

**Comment by datajoely** [comment]
Should we wait until the 0.18.1 fix is out for the templated config loader thing?

**Comment by MerelTheisenQB** [comment]
> Should we wait until the 0.18.1 fix is out for the templated config loader thing?

I think the question is whether this issue will be fixed when the templated config loader problem is solved or whether it's a different problem altogether.

**Comment by AntonyMilneQB** [comment]
This is a different bug and something we should fix ASAP I think. I've done a quick fix here but we should follow up properly: https://github.com/kedro-org/kedro/issues/1437

File: demo-project/src/demo_project/pipelines/reporting/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
-kedro[plotly.PlotlyDataSet, plotly.JSONDataSet]==0.17.6
+kedro[plotly.PlotlyDataSet, plotly.JSONDataSet]=0.18.0
```
Comment (AntonyMilneQB) [inline demo-project/src/demo_project/pipelines/reporting/requirements.txt:1]:
```suggestion
kedro[plotly.PlotlyDataSet, plotly.JSONDataSet]==0.18.0
```

File: demo-project/src/demo_project/requirements.in [hunk 2]
```diff
@@ -1,19 +1,17 @@
-black==21.5b1
+black==22.1.0
```
Comment (AntonyMilneQB) [inline demo-project/src/demo_project/requirements.in:1]:
```suggestion
black~=22.0
```
for consistency with kedro.

**Review by AntonyMilneQB — APPROVED** [review]
Great stuff! It all works nicely on gitpod so I think we're good 🙂

I just pushed a couple of very small changes and left some very minor suggestions.

**Review by rashidakanchwala — APPROVED** [review]
LGTM!

**Pull request merged**